src/SprykerEco/Zed/Unzer/Business/Checkout/Mapper/UnzerCheckoutPostSaveMapper.php

```diff
@@ -57,8 +57,8 @@ public function mapQuoteTransferToUnzerBasketTransfer(
         UnzerBasketTransfer $unzerBasketTransfer
     ): UnzerBasketTransfer {
         return $unzerBasketTransfer
-            ->setAmountTotalGross($quoteTransfer->getTotalsOrFail()->getGrandTotal() / UnzerConstants::INT_TO_FLOAT_DIVIDER)
-            ->setAmountTotalVat($quoteTransfer->getTotalsOrFail()->getTaxTotalOrFail()->getAmount() / UnzerConstants::INT_TO_FLOAT_DIVIDER)
+            ->setAmountTotalGross((int)$quoteTransfer->getTotalsOrFail()->getGrandTotal() / UnzerConstants::INT_TO_FLOAT_DIVIDER)
```

Comment:
are you sure that it should be int? as I remember it should be float here

Reply:
Method getGrandTotal() returns int|null ... If you don't cast return value getGrandTotal() to integer you would get `Left operand cannot be nullable, got int|null`. To fix this we added cast to int which should be imho fine.

Reply:
Fixed.